### PR TITLE
Adds 3 missing circuits to the circuit printer

### PR DESCRIPTION
**Pull request by Dropshotz**
pls don't abuse. Fun new toys to play with tho. Might consider porting bay's circuitry over sometime.

### CODE REVIEW

**Comment by cebutris** [comment]
Unless you can provide a good justification for adding shocker and thrower, we really don't need them back.

Grabber is fine though.

**Comment by Dropshotz** [comment]
Shocker could go maybe, but thrower would still be useful to have around, so long as it's not abused.

EDIT: turned allow edits on, so in case shockers an issue, feel free to yeet that one. Though I don't feel like it's necessary to remove it.

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -203,4 +203,177 @@
...
-	spawn_flags = null			// Used for world initializing, see the #defines above.
\ No newline at end of file
+	spawn_flags = null			// Used for world initializing, see the #defines above.
+
+/obj/item/integrated_circuit/manipulation/shocker
+	name = "shocker circuit"
+	desc = "Used to shock adjastent creatures with electricity."
+	icon_state = "shocker"
+	extended_desc = "The circuit accepts a reference to creature,who needs to be shocked. It can shock target on adjacent tiles. \
+	Severity determines  hardness of shock and it's power consumption. It's given between 0 and 60."
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]:
```suggestion
	Severity determines hardness of shock and its power consumption. It's given between 0 and 60."
```
Its - possesion, it's - it is/it has

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 2]
```diff
@@ -203,4 +203,177 @@
...
+/obj/item/integrated_circuit/manipulation/thrower/New()
+	processing_objects |= src
+/obj/item/integrated_circuit/manipulation/thrower/Destroy()
+	processing_objects -= src
+/obj/item/integrated_circuit/manipulation/thrower/process()
+	set_pin_data(IC_OUTPUT, 1, weakref(contents[1]))
+	set_pin_data(IC_OUTPUT, 2, weakref(contents[contents.len]))
+	set_pin_data(IC_OUTPUT, 3, src.contents.len)
+	push_data()
+*/
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm:266]: Why is this here?

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 3]
```diff
@@ -203,4 +203,177 @@
...
+/obj/item/integrated_circuit/manipulation/grabber/do_work()
+	var/turf/T = get_turf(src)
+	var/atom/movable/AM = get_pin_data_as_type(IC_INPUT, 1, /obj/item)
+	var/mode = get_pin_data(IC_INPUT, 2)
+	if(mode == 1)
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: A comment briefly stating what this mode does would be nice.
```suggestion
	if(mode == 1) // Grab item
```

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 4]
```diff
@@ -203,4 +203,177 @@
...
+			if(contents.len < 10)
+				if(istype(AM,/obj/item))
+					var obj/item/A = AM
+					if(A.w_class < ITEMSIZE_NORMAL)
+						AM.loc = src
+				else if(istype(AM,/mob))
+					var mob/A = AM
+					if(istiny(A))
+						AM.loc = src
+	if(mode == 0)
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: Ditto 271.
```suggestion
	if(mode == 0) // Eject first item
```

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 5]
```diff
@@ -203,4 +203,177 @@
...
+		if(contents.len)
+			var/obj/item/U = contents[1]
+			U.loc = T
+	if(mode == -1)
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: Ditto 271.
```suggestion
	if(mode == -1) // Eject all items
```

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 6]
```diff
@@ -203,4 +203,177 @@
...
+	set_pin_data(IC_OUTPUT, 3, src.contents.len)
+	push_data()
+*/
+/obj/item/integrated_circuit/manipulation/grabber/do_work()
+	var/turf/T = get_turf(src)
+	var/atom/movable/AM = get_pin_data_as_type(IC_INPUT, 1, /obj/item)
+	var/mode = get_pin_data(IC_INPUT, 2)
+	if(mode == 1)
+		if(AM.Adjacent(T))
+			if(contents.len < 10)
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm:273]: Consider condensing this into one statement

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 7]
```diff
@@ -203,4 +203,177 @@
...
+					if(istiny(A))
+						AM.loc = src
+	if(mode == 0)
+		if(contents.len)
+			var/obj/item/U = contents[1]
+			U.loc = T
+	if(mode == -1)
+		if(contents.len)
+			var/obj/item/U
+			for(U in contents)
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm:289]: Should be able to condense this
```suggestion
			for(var/obj/item/U in contents)
```

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 8]
```diff
@@ -203,4 +203,177 @@
...
+	activate_pin(2)
+
+
+
+/obj/item/integrated_circuit/manipulation/thrower
+	name = "thrower"
+	desc = "A compact launcher to throw things from inside or nearby tiles"
+	extended_desc = "The first and second inputs need to be numbers.  They are coordinates to throw thing at, relative to the machine itself.  \
+	The 'fire' activator will cause the mechanism to attempt to throw thing at the coordinates, if possible.  Note that the \
+	projectile need to be inside the machine, or to be on an adjacent tile, and to be up to medium size."
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm:304]: Consider rewording the last half of this line

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 9]
```diff
@@ -203,4 +203,177 @@
...
+	var/turf/TP = get_turf(A)
+	if(!(TP.Adjacent(T)))
+		return
+	if(src.assembly)
+		if(isnum(target_x.data))
+			target_x.data = round(target_x.data)
+		if(isnum(target_y.data))
+			target_y.data = round(target_y.data)
+
+
```
Comment (Shadow-Quill) [inline code/modules/integrated_electronics/subtypes/manipulation.dm:345]:
```suggestion
```
Unneeded whitespace

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]

File: code/modules/integrated_electronics/subtypes/manipulation.dm (hunk 9 above)
Comment (Dropshotz) [reply code/modules/integrated_electronics/subtypes/manipulation.dm:345]: these things came with the code I lifted it from. All I know is they work.

File: code/modules/integrated_electronics/subtypes/manipulation.dm (hunk 7 above)
Comment (Dropshotz) [reply code/modules/integrated_electronics/subtypes/manipulation.dm:289]: did you test this?

File: code/modules/integrated_electronics/subtypes/manipulation.dm (hunk 2 above)
Comment (Dropshotz) [reply code/modules/integrated_electronics/subtypes/manipulation.dm:266]: lifted from previous code
